program-runtime/src/compute_budget.rs

```diff
@@ -127,44 +127,86 @@ impl ComputeBudget {
         message: &SanitizedMessage,
         requestable_heap_size: bool,
         default_units_per_instruction: bool,
+        prioritization_fee_type_change: bool,
     ) -> Result<u64, TransactionError> {
         let mut num_instructions = message.instructions().len();
-        let mut requested_additional_fee = 0;
         let mut requested_units = None;
+        let mut requested_heap_size = None;
+        let mut requested_prioritization_fee = None;
 
-        let error = TransactionError::InstructionError(0, InstructionError::InvalidInstructionData);
         for (i, (program_id, instruction)) in message.program_instructions_iter().enumerate() {
             if compute_budget::check_id(program_id) {
                 // don't include request instructions in default max calc
                 num_instructions = num_instructions.saturating_sub(1);
 
-                // Compute budget instruction must be in the 1st 3 instructions (avoid
-                // nonce marker), otherwise ignored
-                if i < 3 {
-                    match try_from_slice_unchecked(&instruction.data) {
-                        Ok(ComputeBudgetInstruction::RequestUnits {
-                            units,
-                            additional_fee,
-                        }) => {
-                            requested_units = Some(units as u64);
-                            requested_additional_fee = additional_fee as u64;
+                if !prioritization_fee_type_change && i >= 3 {
```

Comment:
I would prefer if this change used this `prioritization_fee_type_change` bool flag only once to branch between the old implementation and the new implementation. It's a bit challenging to follow all the edge cases when the flag is used multiple times. I don't mind a bit of duplicate code here, it's going to be ripped out soon anyways.

Reply:
Alright, dup code wins